src/docs/binding-variables.md

```diff
@@ -0,0 +1,31 @@
+# Binding Variables
+
+Variables in Swift come in two general forms: computed and stored.  Unlike C#, Swift allows variables to exist at the top level of a module being effectively global.
+
+With enable-module-evolution set, all `let` variables will have a getter. Variables that are declared with `var` will have a getter as well as a setter. It is an error to make a set-only variable.
+
+Variables can be:
+- global
+- static
+- instance
+- class (computed only)
+- mutating set (value types only)
+- open (instance in classes only)
+
+## Language Parity Mismatches
+- Naming - Swift can have variable names that are unpronouncable in C#
+- Class - the notion of class variables doesn't exist in C#
+- willSet/didSet - while these can be conceptually modeled with events in C#, these are problematic for binding as there is no real way to override/chain these
```

Comment:
Could these events be represented with the `INotifyPropertyChanging`/`INotifyPropertyChanged` interfaces in `System.ComponentModel` for instance-level (mutating set/open) properties?

Reply:
The problem isn't in C# though, it's in Swift. I had investigated this ~6 years ago and dismissed it as impractical and it created some problems because of (1) implicit chaining of observers and (2) different ordering of the of the chaining (head vs tail). I'll dig out my old notes and add them to the doc

Reply:
If it's too difficult to do, I'm fine with it being marked as out-of-scope.

I think we should mention that these interfaces could be used to model them, but other implementation difficulties make it very expensive (and if we don't need it for SwiftUI or another library that's in scope, then it's out of scope)